From 5219b54cef6afdf01fbf273a232e98f70aa71565 Mon Sep 17 00:00:00 2001
From: Nathan Rossi <nathan@nathanrossi.com>
Date: Mon, 26 Aug 2019 21:03:41 +1000
Subject: [PATCH] Handle tmpnam linker warnings in gcc testsuite

The use of 'tmpnam' is considered dangerous by some libc's (primarily
glibc). The linker warnings are embedded in the libc shared object and
handled by ld (the content of the warning is provided in the shared
object), and cannot be ignored by ld with any command line args.

Treat the warning as acceptable output such that it is not considered as
an excess error, with the use of dg-prune-output.

gcc/testsuite/ChangeLog:

2019-08-26  Nathan Rossi  <nathan@nathanrossi.com>

        * gcc.c-torture/execute/fprintf-2.c: Add dg-prune-output.
        * gcc.c-torture/execute/printf-2.c: Likewise.
        * gcc.c-torture/execute/user-printf.c: Likewise.

Upstream-Status: Pending
---
 gcc/testsuite/gcc.c-torture/execute/fprintf-2.c   | 1 +
 gcc/testsuite/gcc.c-torture/execute/printf-2.c    | 1 +
 gcc/testsuite/gcc.c-torture/execute/user-printf.c | 1 +
 3 files changed, 3 insertions(+)

diff --git a/gcc/testsuite/gcc.c-torture/execute/fprintf-2.c b/gcc/testsuite/gcc.c-torture/execute/fprintf-2.c
index bba4a446ee..5caae12c49 100644
--- a/gcc/testsuite/gcc.c-torture/execute/fprintf-2.c
+++ b/gcc/testsuite/gcc.c-torture/execute/fprintf-2.c
@@ -9,6 +9,7 @@
 
 int main (void)
 {
+// { dg-prune-output "warning: the use of .tmpnam. is dangerous" }
   char *tmpfname = tmpnam (0);
   FILE *f = fopen (tmpfname, "w");
   if (!f)
diff --git a/gcc/testsuite/gcc.c-torture/execute/printf-2.c b/gcc/testsuite/gcc.c-torture/execute/printf-2.c
index 2e9f2a2bb7..0aa5e6d879 100644
--- a/gcc/testsuite/gcc.c-torture/execute/printf-2.c
+++ b/gcc/testsuite/gcc.c-torture/execute/printf-2.c
@@ -23,6 +23,7 @@ write_file (void)
 
 int main (void)
 {
+// { dg-prune-output "warning: the use of .tmpnam. is dangerous" }
   char *tmpfname = tmpnam (0);
   FILE *f = freopen (tmpfname, "w", stdout);
   if (!f)
diff --git a/gcc/testsuite/gcc.c-torture/execute/user-printf.c b/gcc/testsuite/gcc.c-torture/execute/user-printf.c
index 11c61fa394..9f17796de5 100644
--- a/gcc/testsuite/gcc.c-torture/execute/user-printf.c
+++ b/gcc/testsuite/gcc.c-torture/execute/user-printf.c
@@ -20,6 +20,7 @@ user_print (const char *fmt, ...)
 
 int main (void)
 {
+// { dg-prune-output "warning: the use of .tmpnam. is dangerous" }
   char *tmpfname = tmpnam (0);
   FILE *f = freopen (tmpfname, "w", stdout);
   if (!f)
